client/extensions/woocommerce/state/ui/products/edits-reducer.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { isNumber } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	WOOCOMMERCE_EDIT_PRODUCT,
+} from '../../action-types';
+
+const initialState = null;
+
+export default function( state = initialState, action ) {
+	const handlers = {
+		[ WOOCOMMERCE_EDIT_PRODUCT ]: editProductAction,
+	};
+
+	const handler = handlers[ action.type ];
+
+	return ( handler && handler( state, action ) ) || state;
+}
+
+function editProductAction( edits, action ) {
+	const { product, newProductIndex, data } = action.payload;
+
+	return editProduct( edits, product, newProductIndex, data );
+}
+
+function editProduct( edits, product, newProductIndex, data ) {
+	if ( newProductIndex !== undefined ) {
```

Comment:
>  handles newProductIndex property. So presumably, it will be defined.
Actually, we had planned on not passing this for existing products.

>  Why not create separate events WOOCOMMERCE_EDIT_PRODUCT and WOOCOMMERCE_EDIT_NEW_PRODUCT? 

That's a fair point, and one we've discussed. We'll look at separating this out.